src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/ManagedIndexCoordinator.kt

```diff
@@ -315,9 +315,9 @@ class ManagedIndexCoordinator(
     suspend fun getISMTemplates(): Map<String, ISMTemplate> {
         val searchRequest = SearchRequest()
             .source(
-                SearchSourceBuilder().query(
-                    QueryBuilders.existsQuery(ISM_TEMPLATE_FIELD)
-                )
+                SearchSourceBuilder()
+                .query(QueryBuilders.existsQuery(ISM_TEMPLATE_FIELD))
+                .size(MAX_HITS) // Assuming there are not more than 10k policies for now
```

Comment:
Do you think have a cluster setting with default as MAX_HITS instead? in case users want to exceed this or its an overkill for now

Reply:
I think overkill... for one I don't know of anyone really having over 10,000 policies and if a user sets it too low then they could run into the issue that was reported for this. Instead we'll probably have to scroll through and load all the policies, but that's a bit more work and I wanted to get out a quick fix for the user that is affected.